### PR TITLE
KBV-272: Testing GHA with CRON job and adding the KBV Happy path integration test

### PR DESCRIPTION
**Pull request by techaman-m**
## Proposed changes
Add new cucumber steps feature file and its corresponding java implementation for running a pre-merge-integration test with a user identity against the KBV Stub

### What changed

Added `KbvApiHappyPath.feature` and `KbvApiHappyPathSteps.java` for a pre-merge integration test 

### Issue tracking

- [OJ-905](https://govukverify.atlassian.net/browse/OJ-905)
- [OJ-814](https://govukverify.atlassian.net/browse/OJ-814)

### Environment variables or secrets

- API_GATEWAY_ID_PRIVATE
- API_GATEWAY_ID_PUBLIC
- IPV_CORE_STUB_BASIC_AUTH_USER
- IPV_CORE_STUB_BASIC_AUTH_PASSWORD
- IPV_CORE_STUB_URL
- APIGW_API_KEY

Co-authored by: @gokhakan and @jkunle

### CODE REVIEW

File: .github/workflows/pre-merge-integration-test.yml [hunk 1]
```diff
@@ -57,15 +60,28 @@ jobs:
...
+        run: |
+          STACK_NAME=${{ env.STACK_NAME_PREFIX }}-${{ steps.vars.outputs.sha_short }}
+          API_GATEWAY_ID_PRIVATE=$(aws cloudformation describe-stacks --stack-name $STACK_NAME | jq -r '.Stacks[].Outputs[] | select(.OutputKey == "PrivateKBVApiGatewayId").OutputValue')
+          API_GATEWAY_ID_PUBLIC=$(aws cloudformation describe-stacks --stack-name $STACK_NAME | jq -r '.Stacks[].Outputs[] | select(.OutputKey == "PublicKBVApiGatewayId").OutputValue')
+          export API_GATEWAY_ID_PRIVATE=$API_GATEWAY_ID_PRIVATE
+          export API_GATEWAY_ID_PUBLIC=$API_GATEWAY_ID_PUBLIC
+          
+          export IPV_CORE_STUB_BASIC_AUTH_USER=${{ secrets.IPV_CORE_STUB_BASIC_AUTH_USER }}
+          export IPV_CORE_STUB_BASIC_AUTH_PASSWORD=${{ secrets.IPV_CORE_STUB_BASIC_AUTH_PASSWORD }}
+          export IPV_CORE_STUB_URL="https://di-ipv-core-stub.london.cloudapps.digital"
```
Comment (jkunle) [inline .github/workflows/pre-merge-integration-test.yml:83]: Did you try moving these to the `env:`

File: integration-tests/bin/test/features/SheilaHillInsufficientKBVQuestions.feature [hunk 2]
```diff
@@ -0,0 +1,46 @@
+Feature: 3 out of 4 strategy. User Sheila Hil has total of 3 KBV questions.
```
Comment (jkunle) [inline integration-tests/bin/test/features/SheilaHillInsufficientKBVQuestions.feature:1]: Is this a new feature?

**Review by gokhakan — DISMISSED** [review]

**Comment by gokhakan** [comment]
Can we rename the files to `KBVAPIHappyPath.feature` and `KBVCRIHappyPathSteps.java` ?

File: integration-tests/src/test/resources/features/KbvApiHappyPath.feature [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    When user sends a POST request to token end point
+    Then user gets status code 200
+    And a valid access token code is returned in the response
+
+    #Credential Issue
+    When user sends a POST request to Credential Issue end point with a valid access token
+    Then user gets status code 200
+    And a valid JWT is returned in the response
+
+
```
Comment (idrop) [inline integration-tests/src/test/resources/features/KbvApiHappyPath.feature:42]: ☝️ @jkunle @techaman-m @gokhakan this makes me very happy 😀

File: .github/workflows/pre-merge-integration-test.yml [hunk 4]
```diff
@@ -1,5 +1,8 @@
 name: Pre-merge integration test
 on:
+  schedule:
+    # Every weekday at 5am
+    - cron: '0 5 * * 1-5'
```
Comment (idrop) [inline .github/workflows/pre-merge-integration-test.yml:5]: we don't need a cron for pre-merge integration tests

Comment (techaman-m) [reply .github/workflows/pre-merge-integration-test.yml:5]: @gokhakan can you please comment? or can we remove it?

Comment (gokhakan) [reply .github/workflows/pre-merge-integration-test.yml:5]: Agreed.

**Review by gokhakan — APPROVED** [review]

**Review by jkunle — APPROVED** [review]

**Pull request merged**